File: 07.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void *a, const void *b)
{
    return (*(int *)a - *(int *)b);
}

void rearrange(int arr[], int n)
{

    qsort(arr, n, sizeof(int), compare);

    int max = n - 1;
    int min = 0;

    int maxelem = arr[n - 1] + 1;

    for (int i = 0; i < n; i++)
    {
        if (i % 2 == 0)
        {
            arr[i] += (arr[max] % maxelem) * maxelem;
            max--;
        }
        else
        {
            arr[i] += (arr[min] % maxelem) * maxelem;
            min++;
        }
    }

    for (int i = 0; i < n; i++)
    {
        arr[i] = arr[i] / maxelem;
    }

    for (int i = 0; i < n; i++)
    {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

int main()
{
    int n;
    scanf("%d", &n);

    int arr[n];

    for (int i = 0; i < n; i++)
    {
        scanf("%d", &arr[i]);
    }

    rearrange(arr, n);

    return 0;
}

File: 07.py
def rearrange(arr):
    n = len(arr)
    arr.sort()
    max_idx = n - 1
    min_idx = 0
    max_elem = arr[-1] + 1

    for i in range(n):
        if i % 2 == 0:
            arr[i] += (arr[max_idx] % max_elem) * max_elem
            max_idx -= 1
        else:
            arr[i] += (arr[min_idx] % max_elem) * max_elem
            min_idx += 1

    for i in range(n):
        arr[i] = arr[i] // max_elem

    return arr
